src/utils/action.storage.ts

```diff
@@ -1,15 +1,17 @@
+import { UrlParser } from './parsers.js';
+
 /**
  * Store actions as JSON documents on a file system rather than passing huge JSON documents around
  *
  * Uses $ACTION_PATH if set to store the actions to a user defined location
  * This parses $ARGO_TEMPLATE looking for a `archiveLocation`
  */
-export function getActionLocation(): string | null {
-  if (process.env['ACTION_PATH']) return process.env['ACTION_PATH'];
+export async function getActionLocation(): Promise<URL | null> {
```

Comment:
this doesnt need to be made async

you could also use `fsa.toUrl` or make a sync version of the URL parser